pkg/diffs/diffs.go

```diff
@@ -0,0 +1,66 @@
+package diffs
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	prowconfig "k8s.io/test-infra/prow/config"
+)
+
+// Diffs holds the information where the differences can be found.
+type Diffs struct {
+	ConfigPath     string
+	JobConfigPath  string
+	DecorationPath string
+}
+
+// NewDiffs creates a new Diffs{}
+func NewDiffs(configPath, jobConfigPath, decorationPath string) *Diffs {
+	return &Diffs{
+		ConfigPath:     configPath,
+		JobConfigPath:  jobConfigPath,
+		DecorationPath: decorationPath,
+	}
+}
+
+// GetPresubmitsToExecute returns a mapping of repo to presubmits to execute.
+func (d *Diffs) GetPresubmitsToExecute() (map[string][]prowconfig.Presubmit, error) {
+	preSubmitsToExecuteMap := make(map[string][]prowconfig.Presubmit)
+
+	var prJobs map[string]map[string]prowconfig.Presubmit
+	var masterJobs map[string]map[string]prowconfig.Presubmit
+
+	prowMasterConfig, err := prowconfig.Load(d.ConfigPath, d.JobConfigPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load master's Prow config: %v", err)
+	}
+
+	prowPRConfig, err := prowconfig.Load(fmt.Sprintf("%s/cluster/ci/config/prow/config.yaml", d.DecorationPath), fmt.Sprintf("%s/ci-operator/jobs/", d.DecorationPath))
+	if err != nil {
+		return nil, fmt.Errorf("failed to load PR's Prow config: %v", err)
+	}
+
+	masterJobs = getJobsByRepoAndName(prowMasterConfig.JobConfig.Presubmits)
+	prJobs = getJobsByRepoAndName(prowPRConfig.JobConfig.Presubmits)
+	for repo, jobMap := range prJobs {
+		for jobName, job := range jobMap {
+			if !equality.Semantic.DeepEqual(masterJobs[repo][jobName].Spec, job.Spec) {
+				preSubmitsToExecuteMap[repo] = append(preSubmitsToExecuteMap[repo], job)
+			}
+		}
+	}
+	return preSubmitsToExecuteMap, nil
+}
+
+func getJobsByRepoAndName(presubmits map[string][]prowconfig.Presubmit) map[string]map[string]prowconfig.Presubmit {
```

Comment:
Maybe a small comment explaining why this function needs to exist (along the lines of the original thread here https://github.com/openshift/ci-operator-prowgen/pull/50#discussion_r245327669) whould be helpful for the future readers.

Reply:
The function can be inlined if we remove the unnecessary call above.